pkg/distributor/distributor.go

```diff
@@ -218,6 +238,44 @@ func shardByAllLabels(userID string, labels []client.LabelPair) (uint32, error)
 	return h.Sum32(), nil
 }
 
+// Remove the replica label from a slice of LabelPairs if it exists.
+func removeReplicaLabel(replica []byte, labels *[]client.LabelPair) {
+	for i := 0; i < len(*labels); i++ {
+		pair := (*labels)[i]
+		if pair.Name.Equal(replica) {
+			*labels = append((*labels)[:i], (*labels)[i+1:]...)
+			return
+		}
+	}
+}
+
+// Returns an error if we should not accept samples based on the HA labels found in ts. Boolean value
+// is whether we found both HA labels. We use the boolean value to know if we want to remove the
+// replica label.
+func (d *Distributor) checkSample(ctx context.Context, ts client.PreallocTimeseries) (bool, error) {
+	cluster, replica, err := findHALabels([]byte(d.cfg.TrackerConfig.ReplicaLabel), []byte(d.cfg.TrackerConfig.ClusterLabel), ts.Labels)
```

Comment:
This `err` is also never used - might want to consider just not returning it, and using empty strings when you don't find the labels.

Reply:
:+1: